icpc-wf/ansible/roles/cds/tasks/main.yml

```diff
@@ -0,0 +1,139 @@
+---
+# These tasks configure the CDS
+
+- name: Get the latest CDS release
+  uri:
+    url: https://api.github.com/repos/icpctools/icpctools/releases/latest
+    method: GET
+    return_content: yes
+    status_code: 200
+    body_format: json
+  register: latest_cds_release
+
+- name: Set CDS latest version
+  set_fact:
+    cds_version: "{{ latest_cds_release.json.name | replace('v', '') }}"
+
+- name: Check if CDS is installed
+  stat:
+    path: /home/domjudge/cds/wlp/usr/servers/cds/apps/CDS.war
+  register: cds_war
+
+- name: create CDS directory
+  file:
+    path: /home/domjudge/cds
+    state: directory
+    owner: domjudge
+    group: domjudge
+    mode: 0755
+
+- name: Download full CDS ZIP
+  get_url:
+    url: https://github.com/icpctools/icpctools/releases/download/v{{ cds_version }}/wlp.CDS-{{ cds_version }}.zip
+    dest: /root/cds-full.zip
+  when: not cds_war.stat.exists
+
+- name: Unpack full CDS
+  unarchive:
+    src: /root/cds-full.zip
+    dest: /home/domjudge/cds
+    remote_src: true
+    owner: domjudge
+    group: domjudge
+  when: not cds_war.stat.exists
+  notify: restart cds
+
+- name: Download CDS WAR ZIP
+  get_url:
+    url: https://github.com/icpctools/icpctools/releases/download/v{{ cds_version }}/CDS-{{ cds_version }}.zip
+    dest: /root/cds.zip
+  when: cds_war.stat.exists
+
+- name: Unpack CDS WAR
+  unarchive:
+    src: /root/cds.zip
```

Comment:
I think this can be faster?
```
- name: Unarchive a file that needs to be downloaded (added in 2.0)
  unarchive:
    src: https://example.com/example.zip
    dest: /usr/local/bin
    remote_src: yes
```
So you can directly try to feed the URL

Reply:
Ah nice I will try

Reply:
Fixed